plugins/arangodbendorsements/main.go

```diff
@@ -194,67 +186,82 @@ func (e *ArangoEndorsementStore) Init(args common.EndorsementStoreParams) error
 
 	arangoConn := new(ArangoDBConnParams)
 	ctx := context.Background()
-	res, err := e.ConnectToArangoDB(ctx, arangoConn)
-	if (err != nil) || (res != true) {
+	if err := e.ConnectToArangoDB(ctx, arangoConn); err != nil {
 		return fmt.Errorf("Arango DB Connection Failed %v", err)
 	}
 	return nil
 }
 
 // ConnectToArangoDB is responsible for making connection to the arangoDB
-func (e *ArangoEndorsementStore) ConnectToArangoDB(ctx context.Context, arangoConn *ArangoDBConnParams) (connStatus bool, err error) {
-	// Create an HTTP Connection First to the Client
+func (e *ArangoEndorsementStore) ConnectToArangoDB(ctx context.Context, arangoConn *ArangoDBConnParams) error {
+	var ok bool
+	var err error
 
-	ok := true
+	// Create an HTTP Connection First to the Client
 	arangoConn.conn, err = http.NewConnection(http.ConnectionConfig{
 		Endpoints: []string{"http://localhost:8529"},
 	})
 	if err != nil {
-		return false, fmt.Errorf("Failed to create HTTP connection: %v", err)
+		return fmt.Errorf("Failed to create HTTP connection: %v", err)
 	}
 
 	// Create a client
 	arangoConn.client, err = driver.NewClient(driver.ClientConfig{
 		Connection:     arangoConn.conn,
 		Authentication: driver.BasicAuthentication(e.Login, e.Password),
 	})
+	if err != nil {
+		return fmt.Errorf("Failed to create a client: %v", err)
+	}
 
-	// Check if Database with a given name exists or not?
+	// Check if Database with a given name exists or not
 	ok, err = arangoConn.client.DatabaseExists(ctx, e.StoreName)
-	if ok {
-		arangoConn.db, err = arangoConn.client.Database(ctx, e.StoreName)
-		if err != nil {
-			return false, fmt.Errorf("Failed to connect to Endorsement database: %v", err)
+	if err == nil {
+		if ok {
+			arangoConn.db, err = arangoConn.client.Database(ctx, e.StoreName)
+			if err != nil {
+				return fmt.Errorf("Failed to connect to the Endorsement database: %v", err)
+			}
 		}
+		// XXX(tho) -- what if !ok ?
```

Comment:
As this is the Verifier Plugin accessing the Arango Endorsements, Database and Graph should always exist.
Any failure on Database Connection or Graph Connection should result in a failure. Hence please return a failure hereon line 226.